src/Type/BasePickerType.php

```diff
@@ -32,116 +31,205 @@
  */
 abstract class BasePickerType extends AbstractType implements LocaleAwareInterface
 {
+    /**
+     * @var array<string, array<string>|string>
+     */
+    private const DATEPICKER_ALLOWED_OPTIONS = [
```

Comment:
this class is more complicated now because we check ALL the options possible for the date picker, with their types and all.